sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -261,9 +264,28 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         return AccessToken(signature, expiry)
 
 
+if TYPE_CHECKING:
```

Comment:
had to add another TYPE_CHECKING block here, b/c mypy was raising an error here if we included the "EventHubSharedKeyCredential", even with forward referencing.